src/Clean.Architecture.Web/Endpoints/UserEndpoints/SignUp.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Clean.Architecture.Infrastructure.Identity;
+using FastEndpoints;
+using Microsoft.AspNetCore.Identity;
+
+namespace Clean.Architecture.Web.Endpoints.UserEndpoints;
+
+public class SignUp : Endpoint<SignUpUserRequest>
+{
+  private readonly UserManager<User> _userManager;
+  private readonly RoleManager<Role> _roleManager;
+  public SignUp(UserManager<User> userManager, RoleManager<Role> roleManager)
+  {
+    _userManager = userManager;
+    _roleManager = roleManager;
+  }
+
+  public override void Configure()
+  {
+    Post(SignUpUserRequest.Route);
+    AllowAnonymous();
+    Options(x => x
+      .WithTags("UserEndpoints"));
+  }
+  public override async Task HandleAsync(
+    SignUpUserRequest request,
+    CancellationToken cancellationToken)
+  {
+    if (request is null)
+      ThrowError("Request body is empty");
+    //TODO:CHECK FOR USERNAME EMPTY OR NULL
+    var newUser = new User
+    {
+      Age = request.Age,
+      FullName = request.FullName!,
+      Gender = request.Gender,
+      UserName = request.UserName,
+      Email = request.Email
+    };
+    var createUserResult = await _userManager.CreateAsync(newUser, request.Password);
+    if (!createUserResult.Succeeded) { ThrowValidationErrors(createUserResult.Errors); }
+
+    var role = await _roleManager.FindByNameAsync("Admin");
+    if (role==null)
+    {
+      var addRoleResult = await _roleManager.CreateAsync(new Role
+      {
+        Name = "Admin",
+        Description = "admin role"
+      });
+      if (!addRoleResult.Succeeded) { ThrowValidationErrors(addRoleResult.Errors); }
+    }
+    else
```

Comment:
This else does not look like it is necessary.

If I read correctly, you check to see if there is a role by that name already, if it doesn't you create it and only if it does exist do you then assign it to that user in this else clause.

Just removing the else around the inner code looks like it will solve the problem.